### PR TITLE
[To rel/1.1] [IOTDB-5828] Optimize the performance of some parts in metrics, and correcting the metrics count of temporal file size in inner space compaction

### PR DESCRIPTION
**Pull request by THUMarkLau**
See [IOTDB-5828](https://issues.apache.org/jira/browse/IOTDB-5828).

This PR optimize the performance of `JVMThreadMetrics` by caching the result of thread infos, and optimize the performance of `FileMetrics` by counting file in system directory(/proc/[PID]/fd) instead of calling Runtime.exec() in Linux. This PR refactor the way of getting file size and count of temporal files in compaction. The way is changed from updating size incrementally to recording by `CompactionTaskSummary`.

### CODE REVIEW

File: metrics/interface/src/main/java/org/apache/iotdb/metrics/metricsets/jvm/JvmThreadMetrics.java [hunk 1]
```diff
@@ -85,9 +93,25 @@ public void unbindFrom(AbstractMetricService metricService) {
...
+    }
+    lastUpdateTime = System.currentTimeMillis();
+    threadStateCountMap.clear();
+    List<ThreadInfo> infoList =
+        Arrays.asList(threadBean.getThreadInfo(threadBean.getAllThreadIds()));
+    infoList.forEach(
+        info -> {
+          if (info != null) {
+            Thread.State state = info.getThreadState();
+            threadStateCountMap.put(state, threadStateCountMap.getOrDefault(state, 0) + 1);
```
Comment (OneSizeFitsQuorum) [inline metrics/interface/src/main/java/org/apache/iotdb/metrics/metricsets/jvm/JvmThreadMetrics.java:112]: Use the compute API to reduce the number of hashes

File: metrics/interface/src/main/java/org/apache/iotdb/metrics/metricsets/jvm/JvmThreadMetrics.java [hunk 2]
```diff
@@ -25,11 +25,19 @@
...
 import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 /** This file is modified from io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics */
 public class JvmThreadMetrics implements IMetricSet {
+  private static long lastUpdateTime = 0L;
+  private static final long UPDATE_INTERVAL = 10_000L;
+  private static Map<Thread.State, Integer> threadStateCountMap = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (THUMarkLau) [reply metrics/interface/src/main/java/org/apache/iotdb/metrics/metricsets/jvm/JvmThreadMetrics.java:39]: resolve

File: metrics/interface/src/main/java/org/apache/iotdb/metrics/metricsets/jvm/JvmThreadMetrics.java (hunk 1 above)
Comment (THUMarkLau) [reply metrics/interface/src/main/java/org/apache/iotdb/metrics/metricsets/jvm/JvmThreadMetrics.java:112]: resolve

File: server/src/main/java/org/apache/iotdb/db/service/metrics/FileMetrics.java [hunk 3]
```diff
@@ -245,9 +251,17 @@ private void unbindSystemRelatedMetrics(AbstractMetricService metricService) {
   }
 
   private long getOpenFileHandlersNumber() {
```
[the remark this answers is not in the document]
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/service/metrics/FileMetrics.java:253]: resolve

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**